src/dependencies/useStore.ts

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import {
+  Dependency,
+  calculate,
+  calculateForDispatch,
+  makeDependencyIndex,
+  DependencyIndexEntry,
+} from './DependencyMap';
+import { Dispatcher } from 'flux';
+import { get as getDispatcherInstance } from '../dispatcher/DispatcherInstance';
+import { enforceDispatcher } from '../dispatcher/DispatcherInterface';
+import { handleDispatch } from './Dispatch';
+
+type SingleDependency = {
+  [key: string]: Dependency;
+};
+
+function useStore<Props>(
```

Comment:
I think I'd expect this to be called something like `useDependency` rather than `useStore` because you're passing it a DependencyMap. Y'all know the mood in HubSpot better than me though. I think whatever name the most people will understand is best!

Reply:
yeah, i'm very flexible on the name here. i thought about `useDependency` and don't really remember why I settled on `useStore` to be honest. i think you're right that `useDependency` is probably a little more correct. if others feel the same i can rename

Reply:
just had a bit of an internal discussion. i think `useStoreDependency` is what we're gonna go with. `useStore` is technically incorrect, but since `dependency` is a loaded term `useDependency` feels too vague to be helpful. `useStoreDependency` is a bit more verbose but feels both technically correct and unambiguous